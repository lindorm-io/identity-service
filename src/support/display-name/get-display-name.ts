import { IIdentityDisplayName } from "../../entity";
import { IKoaIdentityContext } from "../../typing";
import { findRandomNumber } from "./find-random-number";

export const getDisplayNameObject = (ctx: IKoaIdentityContext) => async (
  displayName: string,
): Promise<IIdentityDisplayName> => {
  const { repository } = ctx;

  const entity = await repository.displayName.findOrCreate({ name: displayName });
  const number = await findRandomNumber({ array: entity.numbers });

  entity.add(number);

  await repository.displayName.update(entity);

  return {
    name: displayName,
    number,
  };
};

export const getDisplayNameString = (displayName: IIdentityDisplayName): string => {
  if (!displayName?.name || !displayName?.number) return null;

  return `${displayName.name}#${displayName.number.toString().padStart(4, "0")}`;
};
