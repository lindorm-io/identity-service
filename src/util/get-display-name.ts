import { isString } from "lodash";
import { IdentityDisplayName } from "../entity";

export const getDisplayName = (displayName: IdentityDisplayName): string =>
  isString(displayName.name) ? `${displayName.name}#${displayName.number.toString().padStart(4, "0")}` : null;
