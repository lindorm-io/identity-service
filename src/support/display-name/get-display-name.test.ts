import MockDate from "mockdate";
import { getDisplayNameObject, getDisplayNameString } from "./get-display-name";
import { getTestRepository, inMemoryStore } from "../../test/grey-box";

jest.mock("uuid", () => ({
  v4: jest.fn(() => "be3a62d1-24a0-401c-96dd-3aff95356811"),
}));
jest.mock("./find-random-number", () => ({
  findRandomNumber: jest.fn(() => 1),
}));

MockDate.set("2020-01-01 08:00:00.000");

describe("getDisplayName", () => {
  let ctx: any;

  beforeEach(async () => {
    ctx = {
      repository: await getTestRepository(),
    };
  });

  test("should return a display name object", async () => {
    await expect(getDisplayNameObject(ctx)("display-name")).resolves.toStrictEqual({ name: "display-name", number: 1 });

    expect(inMemoryStore).toMatchSnapshot();
  });

  test("should return a display name string", async () => {
    expect(getDisplayNameString({ name: "display-name", number: 1 })).toBe("display-name#0001");
  });
});
