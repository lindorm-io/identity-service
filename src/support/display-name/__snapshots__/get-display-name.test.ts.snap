// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDisplayName should return a display name object 1`] = `
Object {
  "identity": Object {
    "display_name": Object {
      "data": Array [
        Object {
          "created": 2020-01-01T07:00:00.000Z,
          "events": Array [
            Object {
              "date": 2020-01-01T07:00:00.000Z,
              "name": "display_name_created",
              "payload": Object {
                "name": "display-name",
                "numbers": Array [
                  1,
                ],
              },
            },
            Object {
              "date": 2020-01-01T07:00:00.000Z,
              "name": "display_name_number_added",
              "payload": Object {
                "number": 1,
              },
            },
          ],
          "id": "be3a62d1-24a0-401c-96dd-3aff95356811",
          "name": "display-name",
          "numbers": Array [
            1,
          ],
          "updated": 2020-01-01T07:00:00.000Z,
          "version": 1,
        },
      ],
      "indices": Array [
        Object {
          "index": Object {
            "id": 1,
          },
          "options": Object {
            "unique": true,
          },
        },
      ],
    },
  },
}
`;
